### PR TITLE
begin by moving methods called in charity api into gem

### PR DESCRIPTION
**Pull request by jesse-spevack**
This pr adds support for all methods called on `Raven` in the charity api to this gem. 

To get this gem working on my machine I had to add `source 'https://rubygems.org'` to the gemfile. I did not commit this change. I'm not sure if that is expected or not.

As a next step, I can start changing charity api from:
```ruby
Raven.breadcrumbs.record do |crumb|
  crumb.message = 'some message'
  crumb.data = { key: 'some value' }
end
```
to: 

```ruby
ExceptionNotifier.breadcrumbs(message: 'some message', data: { key: 'some value' })
```

And I can also change: 

```ruby
# config/initializers/raven.rb

module ActionController
  class Base
    before_action :set_raven_context

    def set_raven_context
      Raven.tags_context(request_id: request.uuid) if request
      Raven.user_context(user_uuid: session[:user_uuid]) if session
      Raven.extra_context(params: params.to_unsafe_h) if params
    end
  end
end
```

to:

```ruby
# config/initializers/raven.rb

module ActionController
  class Base
    before_action :set_exception_notifier_context

    def set_exception_notifier_context
      ExceptionNotifier.add_context(:tags_context, request_id: request.uuid) if request
      ExceptionNotifier.add_context(:user_context, user_uuid: session[:user_uuid]) if session
      ExceptionNotifier.add_context(:extra_context, params: params.to_unsafe_h) if params
    end
  end
end
```

And 
```ruby
let(:breadcrumb) { Raven.breadcrumbs.buffer.last }
```

To
```ruby
let(:breadcrumb) { ExceptionNotifier.last_breadcrumb }
```

### CODE REVIEW

File: lib/exception_notifier.rb [hunk 1]
```diff
@@ -1,30 +1,66 @@
+# frozen_string_literal: true
+
 class ExceptionNotifier
-  def self.call(*args)
-    if exceptionable?(args.first)
-      capture_exception(args)
-    else
-      capture_message(args)
+  CONTEXT_TYPES = %i(extra_context tags_context user_context)
+  class << self
```
Comment (jesse-spevack) [inline lib/exception_notifier.rb:5]: I like this syntactic sugar, but it did result in an awful looking set of diffs.

File: lib/exception_notifier.rb [hunk 2]
```diff
@@ -1,30 +1,66 @@
...
+    end
+
+    def last_breadcrumb
+      error_client.breadcrumbs.buffer.last
+    end
+
+    private
+
+    def error_client
+      Raven
```
Comment (jesse-spevack) [inline lib/exception_notifier.rb:63]: Eventually we'll want to change this when we move to the new sdk.

**Comment by timlawrenz** [comment]
@jesse-spevack This looks pretty promising. Can I ask you to write three basic examples in the readme?

**Comment by jesse-spevack** [comment]
Let me know if there is anything else on here that I need to do to publish the change. Should I manually change the version number in this pr? Once that gets updated, I can then begin a pr in charity api to update the gem version and convert Raven calls to the new api.

**Comment by timlawrenz** [comment]
If this is backwards compatible with the old exception notifier, you should do a minor version update. If the method footprint changed, it's a major version bump. *After that* you can run `gem build exception_notifier.gemspec` and it'll create a new file that should be part of this PR.

**Review by timlawrenz — APPROVED** [review]
this is awesome, thanks! 🤩

**Comment by jesse-spevack** [comment]
Ok, so once this is merged, I can just change: 
```ruby
gem 'exception_notifier', '0.1.3', git: 'https://github.com/givelively/exception_notifier.git'
```
to 
```ruby
gem 'exception_notifier', '0.1.4', git: 'https://github.com/givelively/exception_notifier.git'
```
And convert the old Raven calls to the new exception notifier calls.

**Comment by jesse-spevack** [comment]
🎉

**Pull request merged**